libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -119,12 +120,37 @@ export class LuisRecognizer {
 
     /**
      * Creates a new LuisRecognizer instance.
-     * @param application An object conforming to the [LuisApplication](#luisapplication) definition.
+     * @param application An object conforming to the [LuisApplication](#luisapplication) definition or a string representing a LUIS application endpoint, usually retrieved from https://luis.ai.
      * @param options (Optional) options object used to control predictions. Should conform to the [LuisPrectionOptions](#luispredictionoptions) definition.
      * @param includeApiResults (Optional) flag that if set to `true` will force the inclusion of LUIS Api call in results returned by [recognize()](#recognize). Defaults to a value of `false`.
      */
-    constructor(application: LuisApplication, options?: LuisPredictionOptions, includeApiResults?: boolean) {
-        this.application = application;
+    constructor(application: string, options?: LuisPredictionOptions, includeApiResults?: boolean);
+    constructor(application: LuisApplication, options?: LuisPredictionOptions, includeApiResults?: boolean);
+    constructor(application: LuisApplication|string, options?: LuisPredictionOptions, includeApiResults?: boolean) {
+        if (typeof(application) === 'string') {
+            this.application = {} as LuisApplication;
+            const parsedEndpoint: any = Url(application);
+            if (parsedEndpoint.origin === 'null') {
+                throw new Error(`Unable to parse \`endpoint\` value (e.g. "https://westus.api.cognitive.microsoft.com") from LUIS endpoint ${application}.\nPlease make sure your endpoint is a valid LUIS Endpoint.`);
```

Comment:
Consider a custom Error class here to clean things up:
```js
class EndpointParseError extends Error {
  public static BAD_ORIGIN = 1;
  public static MISSING_APP_ID = 2;
  public static MISSING_ENDPOINT_KEY = 3;
  constructor(type: number, url: string) {
    super();
    this.message = this.getMessage(type, url);
  }
  private getMessage(type: number, url): string {
    switch (type) {
     case EndpointParseError.BAD_ORIGIN:
      return 'a custom message'
     //...
  }
}
//...
// usage
if (parsedEndpoint.origin === 'null') { 
  throw new EndpointParseError(EndpointParseError.BAD_ORIGIN, luisApplication);
}
```

Reply:
i don't think we use custom Error classes elsewhere at this point. I would have thought we can skip this for now.

Reply:
Sure. The constructor in this case is quite messy. This is more about tidying up a bit.